pkg/rpcmiddleware/discover/discover.go

```diff
@@ -52,12 +55,45 @@ func DefaultPort(target string, port int) (string, error) {
 	return target, nil
 }
 
+var clusterRoleServices = map[ttnpb.ClusterRole]string{
+	ttnpb.ClusterRole_ENTITY_REGISTRY:              "is",
+	ttnpb.ClusterRole_ACCESS:                       "is",
+	ttnpb.ClusterRole_GATEWAY_SERVER:               "gs",
+	ttnpb.ClusterRole_NETWORK_SERVER:               "ns",
+	ttnpb.ClusterRole_APPLICATION_SERVER:           "as",
+	ttnpb.ClusterRole_JOIN_SERVER:                  "js",
+	ttnpb.ClusterRole_DEVICE_TEMPLATE_CONVERTER:    "dtc",
+	ttnpb.ClusterRole_DEVICE_CLAIMING_SERVER:       "dcs",
+	ttnpb.ClusterRole_GATEWAY_CONFIGURATION_SERVER: "gcs",
+	ttnpb.ClusterRole_QR_CODE_GENERATOR:            "qrg",
+}
+
 // DefaultPorts is a map of the default gRPC ports, with/without TLS.
 var DefaultPorts = map[bool]int{
 	false: 1884,
 	true:  8884,
 }
 
+var errRole = errors.DefineFailedPrecondition("role", "service role `{role}` not discoverable")
+
+// Scheme returns the gRPC scheme for service discovery.
+func Scheme(role ttnpb.ClusterRole) (string, error) {
+	svc, ok := clusterRoleServices[role]
+	if !ok {
+		return "", errRole.WithAttributes("role", strings.Title(strings.Replace(role.String(), "_", " ", -1)))
+	}
+	return fmt.Sprintf("ttn-v3-%s", svc), nil
+}
+
+// Address returns the host with service discovery gRPC scheme for the given role.
+func Address(role ttnpb.ClusterRole, host string) (string, error) {
+	scheme, err := Scheme(role)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s:///%s", scheme, host), nil
```

Comment:
Why the triple slash? That feels like a mistake, so perhaps good to add a comment here if it is in fact correct.

Reply:
See https://github.com/grpc/grpc/blob/88af358f9ec9b2a84f82b3f4e795df62c920c29b/doc/naming.md; we're not using the authority field.